extras/README.md

```diff
@@ -1,16 +1,97 @@
-# New Actor Template
-Use cargo generate to create a new actor module from this template.
+## How to use  built-in `wasmcloud:extras` capability provider
 
-[Learn how to write](https://wascc.dev/tutorials/first-actor/) waSCC actors, or [explore the actor concept](https://wascc.dev/docs/concepts/actors/).
+`wasmcloud` host runtime comes with some built-in capability providers and `wasmcloud:extras` is one of them. In this tutorial we will generate a random number and retrive some information like request id and number of requests using `wasmcloud:extras`. 
 
-To generate new keys, use `make keys`. This will use `nk` to generate both sets of keys for you, and then write them to the `.keys` directory.
+1. Please install `wasmcloud` and `wash` binaries if you are not already did:
 
-To build your new module, use `make build`. This will compile your code with `cargo`, and then sign it with `wascap` using the keys in `.keys`.
+    https://wasmcloud.dev/overview/installation/
 
-## Tool Requirements
+    ```bash
+    $ wasmcloud --version
+    $ wash --version
+    ```
 
-- Cargo and Rust are required
-- Make is recommended, but not strictly necessary
-- [wascap](https://github.com/wascc/wascap) is required
-- [nk](https://github.com/encabulators/nkeys) is required if you need to generate keys (which you almost certainly do)
+2. Compile and sign the actor:
 
+    ```bash
+    $ cargo build --release
+
+    $ wash claims sign ./target/wasm32-unknown-unknown/release/extras.wasm -c wasmcloud:httpserver -c c wasmcloud:extras --name "extras" --ver 0.2.1
+    ```
+
+    Please note we are signing our actor for multiple capabilities including built-in ones: `wasmcloud:httpserver` and `wasmcloud:extras`. Failing to do so causes mulfunctioning of the actor.
```

Comment:
instead of "causes mulfunctioning of the actor", let's use this as an opportunity to show the strength of deny-by-default security. Maybe something like "Failing to sign our actor with those capabilities will prevent our actor from using them, as wasmcloud uses a deny-by-default security model for capabilities."